### PR TITLE
Reset opt-in status when the election is rescheduled.

### PR DESCRIPTION
**Pull request by swatanabe**
This likely also requires front end updates, as it adds multiple versions to some election states and also changes the meaning of the member::election_participation_status field.

### CODE REVIEW

File: contracts/eden/src/elections.cpp [hunk 1]
```diff
@@ -450,18 +483,15 @@ namespace eden
       {
          if (iter->status() == member_status::active_member)
          {
-            switch (iter->election_participation_status())
+            printf("%d, %d\n", iter->election_participation_status(),
```
Comment (tbfleming) [inline contracts/eden/src/elections.cpp:486]: debugging?

**Comment by tbfleming** [comment]
Since it can automatically bump `election_schedule_version`, this needs an event. Also, eden-micro-chain needs to monitor for the event and clear out members' participating field.

File: contracts/eden/src/actions/elect.cpp [hunk 2]
```diff
@@ -39,7 +39,7 @@ namespace eden
       }
       else
       {
-         eosio::check(member.election_participation_status() == in_election,
+         eosio::check(member.election_participation_status() != not_in_election,
```
Comment (tbfleming) [inline contracts/eden/src/actions/elect.cpp:42]: This check and the one above don't match the new definition of whether the user is opted in

**Comment by swatanabe** [comment]
> Since it can automatically bump election_schedule_version, this needs an event

Should `election_event_schedule` be sufficient?

**Comment by tbfleming** [comment]
> Should `election_event_schedule` be sufficient?

yes

**Review by tbfleming — APPROVED** [review]

**Comment by tbfleming** [comment]
Future PR: subchain needs to clear opt-in on `election_event_schedule`

**Pull request merged**